src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -280,15 +310,74 @@ private boolean parseNodes(List<Node> nodes) {
           continue;
         }
 
-        if (cpeUriCheck(cpeMatch)) {
+        if (projectCheck(cpeMatch, referenceMatch, project)) {
           return true;
         }
       }
 
-      if (parseNodes(node.getChildren())) {
+      if (parseNodes(node.getChildren(), referenceMatch, project)) {
         return true;
       }
     }
     return false;
   }
+
+  /**
+   * Checks if the input {@link URI} is same as the project's URL.
+   * 
+   * @param referenceUrl of type {@link URI}.
+   * @param project of type {@link GitHubProject}.
+   * @return True if the input URL matches the project's URL. Otherwise false.
+   */
+  private static boolean checkUrl(URI referenceUrl, GitHubProject project) {
+    if (referenceUrl.getHost().equals(project.url().getHost())
+        && checkUrlPath(referenceUrl.getPath(), project)) {
+      return true;
+    }
+    return false;
```

Comment:
Could you please simplify it?

```suggestion
    return referenceUrl.getHost().equals(project.url().getHost())
        && checkUrlPath(referenceUrl.getPath(), project);
```